### PR TITLE
Add `extc` functions

### PR DESCRIPTION
**Pull request by adamhutchings**

### CODE REVIEW

File: src/parser/func.c [hunk 1]
```diff
@@ -90,12 +93,25 @@ int yfp_funcdecl(struct yf_parse_node * node, struct yf_lexer * lexer) {
     node->type = YFCS_FUNCDECL;
 
     struct yf_token sctest; /* semicolon test */
+semicolon:
     yfl_lex(lexer, &sctest);
-    if (sctest.type == YFT_SEMICOLON) {
+    switch (sctest.type) {
```
Comment (adamhutchings) [inline src/parser/func.c:98]: @thondascully You're welcome

File: src/api/abstract-tree.h [hunk 2]
```diff
@@ -113,6 +113,7 @@ struct yfa_funcdecl {
     struct yf_ast_node * body; /* The function body */
     /* The parameter scope */
     struct yfs_symtab * param_scope;
+    bool extc; /* Whether the function declaration is extc */
```
Comment (JakeRoggenbuck) [inline src/api/abstract-tree.h:116]: Would this be a good place for a union of struct function and struct external c function?

**Review by thondascully — APPROVED** [review]
Are you sure you want to keep passing a boolean value through the entire interpreting process to only really be used in gen.c? Is this necessary because everything gets interpreted at the same time, so it isn't feasible to separately interpret / pass c-source code? Am just trying to avoid unnecessary if-elses in each step of the entire process.

I am assuming your way right now is the best way to handle at the moment. Nice job. Peer reviewed

File: src/api/abstract-tree.h (hunk 2 above)
Comment (adamhutchings) [reply src/api/abstract-tree.h:116]: Given that the only difference between a normal function and an extc function is a single boolean flag, I'm not sure another way would really be more efficient.

Comment (JakeRoggenbuck) [reply src/api/abstract-tree.h:116]: That makes sense, yea

**Review by JakeRoggenbuck — APPROVED** [review]
LGTM

**Pull request merged**